src/components/common/table/index.tsx

```diff
@@ -28,3 +32,90 @@ export default function Table<T extends TableRowType>(props: TableProps<T>): JSX
     ...(props.showPagination !== false ? { renderPaginationText } : {}),
   });
 }
+
+/**
+ * Ordered columns component that preserves reordered columns using callback functions.
+ * This table fetches column names from preferences and sets them if available.
+ * Also saves new columns order upon reordering, into user preferences (for the org scope).
+ */
+
+export type OrderPreserveableTableProps = {
+  setColumns: (columns: TableColumnType[]) => void;
+};
+
+export function OrderPreserveableTable<T extends TableRowType>(
```

Comment:
this table listens for reorder changes (columns being shuffled) and saves them in the user preference..
on initialization it uses these saved columns (if available) to set the columns order (using a callback - which can have different implementations, species table has its own)